From 096f291a018477eb952302fd0b782392294d530a Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:40:08 -0400
Subject: [ethernet] i40e: (ptp) warn when PF_ID does not match in PRTTSYN_CTL0

Message-id: <1409658042-22689-248-git-send-email-sassmann@redhat.com>
Patchwork-id: 90428
O-Subject: [RHEL7.1 PATCH 247/281] i40e: (ptp) warn when PF_ID does not match in PRTTSYN_CTL0
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jacob Keller <jacob.e.keller@intel.com>

Display a verbose warning message when the incorrect PF attempts to
control timestamping for a port to which it was not assigned. This
shouldn't display except in the case of multiple PFs per port. The
primary intent of this message is to help debugging the reason why the
SIOCSHWTSTAMP ioctl has failed, and to help narrow the cause of the
issue.

Change-ID: Ic98798e0c844d98389d4c20e7160ba256f2bc7e8
Signed-off-by: Jacob Keller <jacob.e.keller@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 302b46449e2918a30b3f98a54653972d2ad0f072)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_ptp.c b/drivers/net/ethernet/intel/i40e/i40e_ptp.c
index 582704a..bb7fe98 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_ptp.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_ptp.c
@@ -447,8 +447,12 @@ static int i40e_ptp_set_timestamp_mode(struct i40e_pf *pf,
  /* Confirm that 1588 is supported on this PF. */
  pf_id = (rd32(hw, I40E_PRTTSYN_CTL0) & I40E_PRTTSYN_CTL0_PF_ID_MASK) >>
   I40E_PRTTSYN_CTL0_PF_ID_SHIFT;
- if (hw->pf_id != pf_id)
-  return -EINVAL;
+ if (hw->pf_id != pf_id) {
+  dev_err(&pf->pdev->dev,
+   "PF %d attempted to control timestamp mode on port %d, which is owned by PF %d\n",
+   hw->pf_id, hw->port, pf_id);
+  return -EPERM;
+ }
 
  switch (config->tx_type) {
  case HWTSTAMP_TX_OFF:
-- 
1.7.1